File: 0988-smallest-string-starting-from-leaf/0988-smallest-string-starting-from-leaf.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    String ans = "";
    public void getString(TreeNode node,String s) {
        char sub = (char)('a' + node.val);
        s = String.valueOf(sub) + s;
        if(node.left == null && node.right == null) {
            if(ans == "") {
                ans = s;
            }
            else {
                if(ans.compareTo(s)>=1) ans = s;
            }
        }
        if(node.left!=null) getString(node.left,s);
        if(node.right!=null) getString(node.right,s);
    }
    public String smallestFromLeaf(TreeNode root) {
        getString(root,"");
        return ans;
    }
}

File: 0988-smallest-string-starting-from-leaf/0988-smallest-string-starting-from-leaf.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    string ans = "";
    
    void getString(TreeNode* node,string s) {
        char sub = (char)('a' + node->val);
        s = sub + s;
        if(node->left == NULL && node->right == NULL) {
            if(ans == "") {
                ans = s;
            }
            else {
                ans = min(s,ans);
            }
        }
        if(node->left!=NULL) getString(node->left,s);
        if(node->right!=NULL) getString(node->right,s);
    }
    string smallestFromLeaf(TreeNode* root) {
        getString(root,"");
        return ans;
    }
};